compact_str/src/to_compact_str_specialisation.rs

```diff
@@ -0,0 +1,166 @@
+use super::CompactStr;
+
+use castaway::{cast, LifetimeFree};
+
+/// # Safety
+///
+///  - CompactStr does not contain any lifetime
+///  - CompactStr is 'static
+///  - CompactStr is a container to `u8`, which is `LifetimeFree`.
+unsafe impl LifetimeFree for CompactStr {}
+
+#[inline(always)]
+pub(super) fn to_compact_str_specialised<T>(val: &T) -> Option<CompactStr> {
+    #[cfg(feature = "to-compact-str-int-spec")]
+    if let Some(compact_str) = int_spec::to_compact_str_specialised(val) {
+        return Some(compact_str);
+    }
+
+    #[cfg(feature = "to-compact-str-float-spec")]
+    // FIXME: The tests for float specialisation fail
+    // on power pc little endian 64-bit:
+    //
+    //     assert_eq!("3.2", 3.2_f32.to_compact_str());
+    //
+    // Expected: 3.2
+    // Actual: 0.0
+    //
+    // It failed only after float specialisation is added.
```

Comment:
Hey, just snooping on this PR as I like to see how Castaway is being used! I'll double-check to make sure that this bug isn't coming from Castaway's side.

Reply:
Awesome!

Since `castaway` is working as intended, I think the bug is more likely to be from the float formatting algorithm in std and ryu.

Seems like they had a hard time running on `powerpc64le` target.

Reply:
Yeah I figured that was the case, but I am beefing up my tests just to be sure. I'm confident in Castaway's logic, but when internally I have a `transmute_unchecked` function because the already-dangerous builtin `transmute` is _too strict_ you can't be too careful. 😀

Reply:
This is confirmed to be a bug in rust as I can reproduce this on my own x86_64 linux box using `cross`, so would have to wait for rust to fix it.